bin/add_celltypes_to_sce.R

```diff
@@ -169,29 +180,30 @@ if (!is.null(opt$singler_results)) {
 
   # add note about cell type method to metadata
   metadata(sce)$celltype_methods <- c(metadata(sce)$celltype_methods, "singler")
+} else {
+  has_singler <- FALSE # use to decide if we should assign consensus labels
 }
 
 # CellAssign results -----------------------------------------------------------
 
-if (!is.null(opt$cellassign_predictions)) {
-  # check that cellassign predictions file was provided
-  if (!file.exists(opt$cellassign_predictions)) {
-    stop("Missing CellAssign predictions file")
-  }
-
-  if (is.null(opt$cellassign_ref_file)) {
-    stop("CellAssign reference filename must be provided")
-  }
+if (file.exists(opt$cellassign_predictions)) {
```

Comment:
Can we simplify some things by checking the file size here?

Then we don't need the `else` block at the end or the internal check where we set predictions as NULL?

```suggestion
has_cellassign <- file.exists(opt$cellassign_predictions) && (file.size(opt$cellassign_predictions) > 0)
if (has_cellassign) {
```